Userland/Libraries/LibJS/Runtime/GlobalObject.cpp

```diff
@@ -359,23 +363,8 @@ Value GlobalObject::get_this_binding(GlobalObject&) const
 // 19.2.1 eval ( x ), https://tc39.es/ecma262/#sec-eval-x
 JS_DEFINE_NATIVE_FUNCTION(GlobalObject::eval)
 {
-    if (!vm.argument(0).is_string())
-        return vm.argument(0);
-    auto& code_string = vm.argument(0).as_string();
-    JS::Parser parser { JS::Lexer { code_string.string() } };
-    auto program = parser.parse_program();
-
-    if (parser.has_errors()) {
-        auto& error = parser.errors()[0];
-        vm.throw_exception<SyntaxError>(global_object, error.to_string());
-        return {};
-    }
-
-    auto& caller_frame = vm.call_stack().at(vm.call_stack().size() - 2);
-    TemporaryChange scope_change(vm.call_frame().environment_record, caller_frame->environment_record);
-
-    auto& interpreter = vm.interpreter();
-    return interpreter.execute_statement(global_object, program).value_or(js_undefined());
+    VERIFY(!vm.in_strict_mode());
```

Comment:
Wait... what? This isn't part of the spec. `(function(){ "use strict"; eval("console.log('hi')") })()` works just fine.